hack/csv-generator.go

```diff
@@ -0,0 +1,238 @@
+	csv := csvv1.ClusterServiceVersion{}
+	err = decoder.Decode(&csv)
+	if err != nil {
+		return err
+	}
+
+	err = replaceVariables(f, &csv)
+	if err != nil {
+		return err
+	}
+
+	marshallObject(csv, os.Stdout)
+	if f.dumpCRDs {
+		files, err := ioutil.ReadDir("data/crd")
+		if err != nil {
+			return err
+		}
+		for _, file := range files {
+			crdFile, err := ioutil.ReadFile(fmt.Sprintf("data/crd/%s", file.Name()))
+			if err != nil {
+				return err
+			}
+
+			crd := extv1beta1.CustomResourceDefinition{}
+			decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(crdFile), 1024)
+			err = decoder.Decode(&crd)
+			if err != nil {
+				return err
+			}
+			marshallObject(crd, os.Stdout)
+		}
+	}
+	return nil
+}
+
+func replaceVariables(flags generatorFlags, csv *csvv1.ClusterServiceVersion) error {
+	csv.Name = "ssp-operator.v" + flags.csvVersion
+	v, err := semver.New(flags.csvVersion)
+	if err != nil {
+		return err
+	}
+	csv.Spec.Version = version.OperatorVersion{
+		Version: *v,
+	}
+
+	for i, container := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Template.Spec.Containers {
+		updatedContainer := container
+		if container.Name == "manager" {
+			updatedContainer.Image = flags.operatorImage
+			updatedVariables := make([]v1.EnvVar, 0)
+			for _, envVariable := range container.Env {
+
+				if envVariable.Name == "KVM_IMAGE" {
+					envVariable.Value = flags.kvmInfoImage
+				}
+				if envVariable.Name == "VALIDATOR_IMAGE" {
+					envVariable.Value = flags.validatorImage
+				}
+				if envVariable.Name == "VIRT_LAUNCHER_IMAGE" {
+					envVariable.Value = flags.virtLauncher
+				}
+				if envVariable.Name == "NODE_LABELLER_IMAGE" {
+					envVariable.Value = flags.nodeLabellerImage
+				}
+				if envVariable.Name == "CPU_PLUGIN_IMAGE" {
+					envVariable.Value = flags.cpuPlugin
+				}
+				if envVariable.Name == "OPERATOR_VERSION" {
+					envVariable.Value = flags.operatorVersion
+				}
+				updatedVariables = append(updatedVariables, envVariable)
+			}
+			updatedContainer.Env = updatedVariables
+			csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Spec.Template.Spec.Containers[i] = updatedContainer
+			break
+		}
+	}
+	return nil
+}
+
+func marshallObject(obj interface{}, writer io.Writer) error {
+	jsonBytes, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	var r unstructured.Unstructured
+	if err := json.Unmarshal(jsonBytes, &r.Object); err != nil {
+		return err
+	}
+
+	// remove status and metadata.creationTimestamp
+	unstructured.RemoveNestedField(r.Object, "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "spec", "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "status")
+
+	deployments, exists, err := unstructured.NestedSlice(r.Object, "spec", "install", "spec", "deployments")
```

Comment:
The `err` returned value from this line is never checked. If an error can never happen, please use `_` to ignore the returned value.